packages/hyperion-autologging/src/ALSurfaceVisibilityPublisher.ts

```diff
@@ -119,29 +124,137 @@ export function publish(options: InitOptions): void {
     }
   }
 
-  function observe(surfaceData: ALSurfaceData, element: Element, trackVisibilityThreshold: number): void {
-    const observer = getOrCreateObserver(trackVisibilityThreshold);
-    /**
-     * IntersectionObserver cannot track display:content styles because
-     * these elements don't have their own "box".
-     * So, instead we have to focus on the children of the parent element
-     */
+  // Helper functions for tracking and untracking elements
+  function trackElement(observer: IntersectionObserver, element: Element, surfaceData: ALSurfaceData, observeElement: boolean = true): void {
+    if (observeElement) {
+      observer.observe(element);
+    }
+    setVirtualPropertyValue<IntersectionObserver>(
+      element,
+      VISIBILITY_OBSERVER_PROP,
+      observer,
+    )
+    setVirtualPropertyValue<ALSurfaceData>(
+      element,
+      SURFACE_DATA_PROP,
+      surfaceData,
+    )
+    observedRoots.set(element, surfaceData);
+    surfaceDataRoots.set(surfaceData, element);
+  }
+
+  function untrackElement(observer: IntersectionObserver, element: Element, surfaceData: ALSurfaceData): void {
+    observer.unobserve(element);
+    observedRoots.delete(element, surfaceData);
+    surfaceDataRoots.delete(surfaceData, element);
```

Comment:
I think you should also remove the virtual property values, for example setting them as null.